host/src/server/api/v3/proof/mod.rs

```diff
@@ -0,0 +1,171 @@
+use std::str::FromStr;
+
+use anyhow::anyhow;
+use axum::{debug_handler, extract::State, routing::post, Json, Router};
+use raiko_core::{
+    interfaces::{AggregationRequest, ProofRequest, ProofRequestOpt, ProofType},
+    provider::get_task_data,
+};
+use raiko_lib::input::{AggregationGuestInput, AggregationGuestOutput};
+use raiko_tasks::{TaskDescriptor, TaskManager, TaskStatus};
+use reth_primitives::B256;
+use utoipa::OpenApi;
+
+use crate::{
+    interfaces::HostResult,
+    metrics::{inc_current_req, inc_guest_req_count, inc_host_req_count},
+    server::api::{v2, v3::Status},
+    Message, ProverState,
+};
+
+mod cancel;
+
+#[utoipa::path(post, path = "/proof",
+    tag = "Proving",
+    request_body = ProofRequestOpt,
+    responses (
+        (status = 200, description = "Successfully submitted proof task, queried tasks in progress or retrieved proof.", body = Status)
+    )
+)]
+#[debug_handler(state = ProverState)]
+/// Submit a proof aggregation task with requested config, get task status or get proof value.
+///
+/// Accepts a proof request and creates a proving task with the specified guest prover.
+/// The guest provers currently available are:
+/// - native - constructs a block and checks for equality
+/// - sgx - uses the sgx environment to construct a block and produce proof of execution
+/// - sp1 - uses the sp1 prover
+/// - risc0 - uses the risc0 prover
+async fn proof_handler(
+    State(prover_state): State<ProverState>,
+    Json(mut aggregation_request): Json<AggregationRequest>,
+) -> HostResult<Status> {
+    inc_current_req();
+    // Override the existing proof request config from the config file and command line
+    // options with the request from the client.
+    aggregation_request.merge(&prover_state.request_config())?;
+
+    let mut tasks = Vec::with_capacity(aggregation_request.block_numbers.len());
+
+    let proof_request_opts: Vec<ProofRequestOpt> = aggregation_request.clone().into();
+
+    // Construct the actual proof request from the available configs.
+    for proof_request_opt in proof_request_opts {
+        let proof_request = ProofRequest::try_from(proof_request_opt)?;
+
+        inc_host_req_count(proof_request.block_number);
+        inc_guest_req_count(&proof_request.proof_type, proof_request.block_number);
+
+        let (chain_id, blockhash) = get_task_data(
+            &proof_request.network,
+            proof_request.block_number,
+            &prover_state.chain_specs,
+        )
+        .await?;
+
+        let key = TaskDescriptor::from((
+            chain_id,
+            blockhash,
+            proof_request.proof_type,
+            proof_request.prover.to_string(),
+        ));
+
+        tasks.push(key);
+    }
+
+    let mut manager = prover_state.task_manager();
+
+    let mut is_registered = false;
+    let mut is_success = true;
+
+    for task in tasks.iter() {
```

Comment:
feel like this is a burst resource consumption (network/cpu/mem), do you think the streaming style is better than this? i.e: 
`start-1 ..general 12s interval.. 3 ..interval.. 5 .. 7-end .`
Pros is no burst, and resource consumption of single proof task stays in same level as before (if aggregation does not require lots of resources)
Cons is complex status management, lots of corner cases I guess.

Reply:
I do believe that the streaming style will be better. How difficult is it for the client to work with the streaming approach?

Reply:
I think the complexity comes from both the API and internal status management. API is kind of negotiation design between us raiko & client: a general example be like client asks 1,2,3,4,5, but 3 failed, we then have to generate [1,2,4,5] proofs and send the aggregation proof back with [1,2,4,5] block numbers, status management issues includes how long we keep those individual proofs (avoid disk overflow), if persistent intermedia proof storage is needed (avoid requests lost after raiko restart), etc.